apps/ledger-live-desktop/src/renderer/modals/SignMessage/steps/StepSummary.tsx

```diff
@@ -60,43 +63,88 @@ const AdvancedMessageArea = styled.pre`
   padding: 10px;
 `;
 
-const MessageProperty = memo(({ label, value }: MessageProperties[0]) => {
-  if (!value) return null;
-
-  return (
-    <Box flex="1" mb={20}>
-      <Text ff="Inter|Medium" color="palette.text.shade40" fontSize={4}>
-        {label}
-      </Text>
-      <Text ff="Inter|Medium" color="palette.text.shade90" fontSize={3} pl={2}>
-        {typeof value === "string" ? (
-          <ValueWrapper>{value}</ValueWrapper>
-        ) : (
-          <PropertiesList>
-            {value?.map?.((v, i) => (
-              <li key={i}>
-                <ValueWrapper>{`${v}${i < value.length - 1 ? "," : ""}`}</ValueWrapper>
-              </li>
-            ))}
-          </PropertiesList>
-        )}
-      </Text>
-    </Box>
-  );
-});
+const MessageProperty = memo(
+  ({
+    label,
+    value,
+    account,
+    contractAddress,
+  }: MessageProperties[0] & { account?: Account; contractAddress?: string | string[] }) => {
+    if (!value) return null;
+
+    const isPropertyAmount = label.includes("Amount") && account && contractAddress;
+
+    const tokenUnit = isPropertyAmount
+      ? account.subAccounts?.find(a => a.token.contractAddress === contractAddress)?.token.units[0]
+      : undefined;
+
+    return (
+      <Box
+        style={{ flexDirection: "row", justifyContent: "space-between", maxWidth: "100%" }}
+        flex="1"
+        mb={20}
+      >
+        <Text ff="Inter|Medium" color="palette.text.shade40" fontSize={4}>
+          {label}
+        </Text>
+        <Text
+          ff="Inter|Medium"
+          color="palette.text.shade90"
+          fontSize={3}
+          pl={2}
+          style={{ maxWidth: "90%" }}
+        >
+          {typeof value === "string" ? (
+            <ValueWrapper>
+              {tokenUnit ? (
+                <>
+                  <FormattedVal
+                    color={"palette.neutral.c100"}
+                    val={Number(value)}
+                    unit={tokenUnit}
+                    fontSize={3}
+                    disableRounding
+                    alwaysShowValue
+                    showCode
+                    inline
+                  />
+                </>
```

Comment:
Same as the other one, not sure why we have a fragment here and `FormattedVal` is usually used outside a `Text` component